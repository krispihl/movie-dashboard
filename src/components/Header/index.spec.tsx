import { Header } from "./index";
import { render, screen } from '@testing-library/react';

jest.mock('../../contexts/movie-details', () => ({
    useMovieDetails: () => jest.fn(),
}));

jest.mock('../Search', () => ({
    Search: () => {
        return <div className='mocked-search' />
    }
}));

jest.mock('../Filter', () => ({
    Filter: () => {
        return <div className='mocked-filters' />
    }
}));

describe('src/components/Header', () => {
    it('should render Header correctly', async () => {
        const { container } = render(<Header />);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Popular movies')).toBeInTheDocument();
    });
});
