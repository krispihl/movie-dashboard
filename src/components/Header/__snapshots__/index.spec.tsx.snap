// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Header should render Header correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__row header__row--top"
    >
      <h1
        class="header__title"
      >
        Popular movies
      </h1>
      <div
        class="mocked-search"
      />
    </div>
    <div
      class="header__row header__row--bottom"
    >
      <div
        class="mocked-filters"
      />
    </div>
  </header>
</div>
`;
